lib/logstash/plugin.rb

```diff
@@ -59,6 +59,11 @@ def inspect
     end
   end
 
+  def dead_letter(event)
+    return unless event.is_a?(LogStash::Event)
+    LogStash::DeadLetterPostOffice << event
```

Comment:
I'm not sure this needs to be global. We already provide the 'logger' to each plugin instance, and we could provide a DeadLetter::Destination to each plugin instance at creation-time.

Thoughts? My preference is to try and avoid global singletons like this so that we can more easily test via dependency injection and also perhaps have more than one DeadLetteRPostOffice in a single program.